### PR TITLE
OH2-184 | Update visits filter

### PR DESCRIPTION
**Pull request by SilverD3**
Updates :

- Age min (0) and max (200);
- Disable future date;
- Set `month`, `dateFrom` and `dateTo` based on selected `year`.

### CODE REVIEW

File: src/components/accessories/opds/filter/OpdFilterForm.tsx [hunk 1]
```diff
@@ -83,27 +91,65 @@ export const OpdFilterForm: FC<IOpdFilterProps> = ({ fields, onSubmit }) => {
...
+        test: function (value) {
+          return !value || (value && value >= 0);
+        },
+      })
+      .test({
+        name: "maxAgeFrom",
+        message: t("opd.abovemaxage"),
+        test: function (value) {
+          return !value || (value && value <= 200);
+        },
```
Comment (axelpezzo) [inline src/components/accessories/opds/filter/OpdFilterForm.tsx:151]: I think that we should change the name from "From" to "To"

**Comment by axelpezzo** [comment]
@SilverD3 Can you explain what's the improvement in this PR about the month / year fields? I mean, I can't understand completely what you did and why... maybe could you plese explain with some example?

**Comment by SilverD3** [comment]
> @SilverD3 Can you explain what's the improvement in this PR about the month / year fields? I mean, I can't understand completely what you did and why... maybe could you plese explain with some example?
1. Why i did the task
The issue was raised by Alberto during a meeting. And @tsognong created the ticket [OH2-184](https://openhospital.atlassian.net/browse/OH2-184) to solve the issue.
2. The improvements
As described in the ticket:

- I've set the min. age to 0 and max. age to 200;
- I've disabled filtering OPDs by future date (invalidate form when future date is entered);
- Make form to consider selected year (currently, current year is always considered, no matter of selected one) when the user choose a month.

File: src/components/accessories/opds/filter/OpdFilterForm.tsx (hunk 1 above)
Comment (SilverD3) [reply src/components/accessories/opds/filter/OpdFilterForm.tsx:151]: Absolutely. `Copy and paste` effect :)

**Comment by axelpezzo** [comment]
Hello @SilverD3 could you please fix those typo in the OpdFilterForm file?

**Comment by SilverD3** [comment]
> Hello @SilverD3 could you please fix those typo in the OpdFilterForm file?

Yes. I'm currently working in another task (another branch) and I have uncommitted changes. So I have to finish the task and commit all those changes.

**Review by axelpezzo — APPROVED** [review]

**Pull request merged**